src/go/k8s/pkg/resources/configuration/patch.go

```diff
@@ -76,6 +79,24 @@ func ThreeWayMerge(
 	}
 	for k, v := range apply {
 		if oldValue, ok := current[k]; !ok || !PropertiesEqual(log, v, oldValue, schema[k]) {
+			// Value must be type []interface{} if array
+			// https://github.com/redpanda-data/team-kubernetes-internal/issues/11
+			if meta, ok := schema[k]; ok && meta.Type == "array" {
+				value, ok := v.(string)
+				if !ok {
+					log.V(debugLogLevel).Info(fmt.Sprintf("computing patch: expected value to be string, got %T", v), "key", k, "value", v)
+					continue
+				}
+				var a []interface{}
+				if err := yaml.Unmarshal([]byte(value), &a); err != nil {
+					log.Error(err, fmt.Sprintf("computing patch: unable to marshal config to %T", a), "key", k, "value", value)
+					// want: fail fast, admin api will eventually return error
```

Comment:
This comment is misleading. It says fail fast, but it is still sends `v` to patch. I think this comment should be changed that if value can not be unmarshal as array, then content of the value will be passed as it was provided by the user.

Reply:
what I meant in a `want` comment is that it is not exactly `todo` but nice-to-have
> if value can not be unmarshal as array, then content of the value will be passed as it was provided by the user

this is what we are doing here